src/hooks/useTimer.ts

```diff
@@ -0,0 +1,32 @@
+import { useState, useEffect } from 'react';
+
+export function useTimer(initialTime: number) {
```

Comment:
타이머에 callback 함수를 인자로 넘겨받는건 어떠세요?
```suggestion
export function useTimer(initialTime: number, callback: ()=>void) {
```
타이머가 끝났을 때 실행할 로직을 함수로 넘겨주면 깔끔할 것 같습니다.

Reply:
이렇게되면, 버튼의 disable 처리하는걸 상태로 빼서 setState를 처리하는 함수를 콜백으로 넘겨도 됩니다.
나중에 프로젝트 연동하기 할 때도 쓰일 것으로 예상되어, 콜백 prop 하나 추가되면 좋을 것 같아요

Reply:
`callback: () => void` 라면 setTimerId(null); 다음 그냥 `callback();` 이면 될 것 같고, `callback?: () => void`과 같이 옵셔널이라면 아래 작성해주신 것처럼 조건문으로 걸면 될 것 같습니다. 콜백 함수를 선택으로 두고 싶으셔서 아래 조건문을 거신건가요?

Reply:
아 네네 그냥 급하게 테스트 해보느라 `?` 적었습니다
필수값으로 두고 `if(callback)` 조건 제거해도 될 것 같습니다.

Reply:
그럼 선택으로 추가하겠습니다!

Reply:
수정했습니다!